pkg/converters/ingress/annotations/backend.go

```diff
@@ -61,7 +61,26 @@ func (c *updater) buildBackendAffinity(d *backData) {
 	}
 	d.backend.Cookie.Keywords = keywords
 	d.backend.Cookie.Dynamic = d.mapper.Get(ingtypes.BackSessionCookieDynamic).Bool()
+	d.backend.Cookie.Preserve = d.mapper.Get(ingtypes.BackSessionCookiePreserve).Bool()
 	d.backend.Cookie.Shared = d.mapper.Get(ingtypes.BackSessionCookieShared).Bool()
+
+	if strings.Contains(d.backend.Cookie.Keywords, "preserve") {
+		// just warn, no error, for keeping backwards compatibility where "preserve" may have been used in the "keywords" section
+		c.logger.Warn("session-cookie-keywords contains 'preserve'; consider using 'session-cookie-preserve' instead for better dynamic update cookie persistence")
```

Comment:
Add the source of the annotation here, something like "session-cookie-keywords on %s contains 'preserve'...". `d.mapper.Get(...).Source` has the source in string format and `%s` does not need to be quoted. This way one can easily find its ingress or service on a large k8s cluster. It'd suggest also to extract to a local var and use `.Bool()` and `Source` from there.

Reply:
Implemented in https://github.com/jcmoraisjr/haproxy-ingress/pull/667/commits/474d3c5f9bb044346cef58112f6e557844178696 

By `.Bool()` I am thinking you meant just `.Value` as it's a string value. Let me know if I misinterpreted that part.

Reply:
Sure, yes, `.Value`. I did a little mess with `session-cookie-preserve` which would use `.Bool()`.